pkg/gadgets/socket-collector/tracer/tracer.go

```diff
@@ -0,0 +1,194 @@
+	ip := net.IP(ipBytes)
+
+	return ip.String()
+}
+
+// Parse is done based on what is defined kernel: https://github.com/torvalds/linux/blob/v5.13/include/net/tcp_states.h#L12-L24
+func parseStatus(proto string, statusUint uint8) (string, error) {
+	statusMap := [...]string{"ESTABLISHED", "SYN_SENT", "SYN_RECV", "FIN_WAIT1", "FIN_WAIT2", "TIME_WAIT", "CLOSE", "CLOSE_WAIT", "LAST_ACK", "LISTEN", "CLOSING", "NEW_SYN_RECV"}
+
+	// Kernel enum starts from 1, adjust it to the statusMap
+	statusUint--
+
+	if len(statusMap) < int(statusUint) {
+		return "", fmt.Errorf("invalid %s status: %d", proto, statusUint+1)
+	}
+
+	status := statusMap[statusUint]
+
+	// Transform TCP status into something more suitable for UDP
+	if proto == "udp" {
+		if status == "ESTABLISHED" {
+			status = "ACTIVE"
+		} else if status == "CLOSE" {
+			status = "INACTIVE"
+		} else {
+			return "", fmt.Errorf("unexpected %s status %s", proto, status)
+		}
+	}
+
+	return status, nil
+}
+
+func getTCPIter() (*link.Iter, error) {
+	objs := IterTCPv4Objects{}
+	if err := LoadIterTCPv4Objects(&objs, nil); err != nil {
+		return nil, fmt.Errorf("failed to load TCP BPF objects: %v", err)
+	}
+	defer objs.Close()
+
+	it, err := link.AttachIter(link.IterOptions{
+		Program: objs.DumpTcp4,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to attach TCP BPF iterator: %v", err)
+	}
+
+	return it, nil
+}
+
+func getUDPIter() (*link.Iter, error) {
+	objs := IterUDPv4Objects{}
+	if err := LoadIterUDPv4Objects(&objs, nil); err != nil {
+		return nil, fmt.Errorf("failed to load UDP BPF objects: %v", err)
+	}
+	defer objs.Close()
+
+	it, err := link.AttachIter(link.IterOptions{
+		Program: objs.DumpUdp4,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to attach UDP BPF iterator: %v", err)
+	}
+
+	return it, nil
+}
+
+func RunCollector(pid uint32, filter *gadgetv1alpha1.ContainerFilter) (string, error) {
+	var err error
+	var it *link.Iter
+	iters := []*link.Iter{}
+
+	// TODO: Use filters to collect only TCP or only UDP information
+	it, err = getTCPIter()
+	if err != nil {
+		return "", err
+	}
+	iters = append(iters, it)
+
+	it, err = getUDPIter()
+	if err != nil {
+		return "", err
+	}
+	iters = append(iters, it)
+
+	// Lock the OS Thread so we don't accidentally switch namespaces. Further details in:
+	// https://github.com/vishvananda/netns/blob/2eb08e3e575f00733a612d25cc5d7470f8db6f35/README.md
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	// TCP/UDP BPF Iters read information from the current namespace thus
+	// we need to change network namespace before opening the iterator
+	netnsHandle, err := netns.GetFromPid(int(pid))
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve network namespace from PID %d: %v", pid, err)
+	}
+	defer netnsHandle.Close()
+
+	if err := netns.Set(netnsHandle); err != nil {
+		return "", fmt.Errorf("error changing network namespace of PID %d: %v", pid, err)
+	}
+
+	var sockets []Socket
```

Comment:
```suggestion
	sockets := []Socket{}
```

See https://play.golang.org/p/DhZfxZAyCnM to see the difference produced in the marshalled output